File: 07282023/50_167_List_Manipulation/python/list_manipulation.py
def list_manipulation(a_list: list, command: str, location: str, value=None):
    """Takes in commands and a list and returns modified list depending on command."""
    if (command=="remove" and location=="end"):
        return a_list.pop()
    elif (command=="remove" and location=="beginning"):
        return a_list.pop(0)
    elif (command=="add" and location=="beginning"):
        a_list.insert(0, value)
    elif (command=="add" and location=="end"):
        a_list.append(value)
    return a_list


print(list_manipulation([1,2,3], "remove", "end")) # 3
print(list_manipulation([1,2,3], "remove", "beginning")) #  1
print(list_manipulation([1,2,3], "add", "beginning", 20)) #  [20,1,2,3]
print(list_manipulation([1,2,3], "add", "end", 30)) #  [1,2,3,30]

File: 07282023/50_167_List_Manipulation/java/ListManipulation.java
import java.util.List;
import java.util.Arrays;
import java.util.ArrayList;

public class ListManipulation {
    public static void main(String[] args) {
        List<Integer> myList1 = new ArrayList<>(Arrays.asList(1, 2, 3));
        List<Integer> myList2 = new ArrayList<>(Arrays.asList(1, 2, 3));
        List<Integer> myList3 = new ArrayList<>(Arrays.asList(1, 2, 3));
        List<Integer> myList4 = new ArrayList<>(Arrays.asList(1, 2, 3));

        System.out.println(listManipulation(myList1, "remove", "end", null));
        System.out.println(listManipulation(myList2, "remove", "beginning", null));
        System.out.println(listManipulation(myList3, "add", "beginning", 20));
        System.out.println(listManipulation(myList4, "add", "end", 30));
    }

    public static Object listManipulation(List<Integer> aList, String command, String location, Integer value) {
        if (command.equals("remove") && location.equals("end")) {
            Integer removedElement = aList.remove(aList.size() - 1);
            return removedElement;
        } else if (command.equals("remove") && location.equals("beginning")) {
            Integer removedElement = aList.remove(0);
            return removedElement;
        } else if (command.equals("add") && location.equals("beginning")) {
            aList.add(0, value);
        } else if (command.equals("add") && location.equals("end")) {
            aList.add(value);
        }
        return aList;
    }
}